modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/utils/ServerUtilities.java

```diff
@@ -0,0 +1,375 @@
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+
+			} catch (ClientProtocolException e) {
+				message = "Invalid client protocol.";
+				throw new IDPTokenManagerException(message, e);
+			} catch (IOException e) {
+				message = "Server connectivity failure.";
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+				                   Constants.INTERNAL_SERVER_ERROR);
+				throw new IDPTokenManagerException(message, e);
+			}
+		} else if (httpMethod.equals(Constants.GET_METHOD)) {
+			HttpGet httpGet = new HttpGet(url);
+			HttpGet httpGetWithHeaders = (HttpGet) buildHeaders(httpGet, headers, httpMethod);
+
+			try {
+				HttpResponse response = httpClient.execute(httpGetWithHeaders);
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+				                   String.valueOf(response.getStatusLine().getStatusCode()));
+			} catch (ClientProtocolException e) {
+				message = "Invalid client protocol.";
+				throw new IDPTokenManagerException(message, e);
+			} catch (IOException e) {
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, "Internal Server Error");
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS,
+				                   Constants.INTERNAL_SERVER_ERROR);
+				message = "Server connectivity failure.";
+				throw new IDPTokenManagerException(message, e);
+			}
+		}
+
+		return responseParams;
+
+	}
+
+	public static Map<String, String> postDataAPI(APIUtilities apiUtilities,
+	                                              Map<String, String> headers)
+			throws IDPTokenManagerException {
+		String httpMethod = apiUtilities.getHttpMethod();
+		String message;
+		String url = apiUtilities.getEndPoint();
+		Map<String, String> params = apiUtilities.getRequestParamsMap();
+
+		Map<String, String> responseParams = new HashMap<String, String>();
+		HttpClient httpclient = getCertifiedHttpClient();
+		String payload = buildPayload(params);
+
+		if (httpMethod.equals(Constants.POST_METHOD)) {
+			HttpPost httpPost = new HttpPost(url);
+			HttpPost httpPostWithHeaders = (HttpPost) buildHeaders(httpPost, headers, httpMethod);
+			byte[] postData = payload.getBytes();
+			try {
+				httpPostWithHeaders.setEntity(new ByteArrayEntity(postData));
+				HttpResponse response = httpclient.execute(httpPostWithHeaders);
+				String status = String.valueOf(response.getStatusLine().getStatusCode());
+
+				responseParams.put(Constants.SERVER_RESPONSE_BODY, getResponseBody(response));
+				responseParams.put(Constants.SERVER_RESPONSE_STATUS, status);
+				return responseParams;
+			} catch (ClientProtocolException e) {
+				message = "Invalid client protocol.";
```

Comment:
Here, the assigning the log message to a variable called "message" is redundant, I believe. Instead, you can directly pass the message into the Exception instance.